### DISCUSSION TITLE
Type error on name parameter

### DISCUSSION THREAD

**Discussion by tvanrijn** [react-hook-form/react-hook-form#5329] [Q&A] [answered]
**Describe the bug**
Type `string` is not supported on the name parameter of the Controller component, typescript gives the error: 
```
Type 'string' is not assignable to type '`${string}` | `${string}.${string}` | `${string}.${number}`'
```

**To Reproduce**

1. Go to https://codesandbox.io/s/goofy-ptolemy-4y3p0
2. Open file components/Input.tsx
3. Scroll down to line 14
4. See the type error

**Codesandbox link (Required)**
https://codesandbox.io/s/goofy-ptolemy-4y3p0?file=/src/components/Input.tsx

**Expected behavior**
I would expect type string to be allowed.

**Additional context**
Using Typescript 4.1.5

<img width="515" alt="image" src="https://user-images.githubusercontent.com/12894463/119488677-1d871e00-bd5b-11eb-8878-ae52b8c8719e.png">

**Comment by bluebill1049** [comment, marked as answer]
In your codesandbox, looks like the generic is missing at `useForm` which is required for hook form to type check.

https://react-hook-form.com/get-started#TypeScript

**Reply by tvanrijn** [reply]
Thanks for the fast reply! I've added the type to useForm, but that doesn't seem to solve the problem.

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/infallible-grass-jfih0?file=/src/components/Form.tsx:131-139

**Reply by bluebill1049** [reply]
Let me know if above helps.

**Reply by tvanrijn** [reply]
got it, thanks!
